### PR TITLE
Store registered plant on namespace

### PR DESCRIPTION
**Pull request by bas080**
Handy for when modders want to use these definitions to extend or add to the mod

### CODE REVIEW

**Comment by sofar** [comment]
I can see this being useful to extend the existing plants.

**Comment by bas080** [comment]
I am working on making a mod that makes farming mod plants spawn in the wild. It depends on VanessaE's biome_lib. Unless biome_lib becomes part of minetest_game i think it is best to keep the wild versions of the plants as a separate mod while still leveraging the existing definitions.

**Comment by rubenwardy** [comment]
:+1: seems like this could be useful, and is a small change

**Comment by bas080** [comment]
@rubenwardy, I have not considered that approach. I might also understand your hesitance to allow mods to do this as the table might not be there forever. It creates quite the dependency.

The counter argument being that the mod that i'm making is completely about these farm plants. If these plants change or if the table is removed the mod should break.

**Comment by paramat** [comment]
> Unless biome_lib becomes part of minetest_game

It won't.

**Comment by rubenwardy** [comment]
I can't see any harm in this. Use cases:

1. copying and making a new plant
2. get / override the min and max light that the plant grows in
3. get the number of steps in growth without having to guess by checking for the biggest node registered in the pattern `farming:pname_[0-9]+`.
4. looping through all registered plants (already possible using groups)

You can't override the steps using the saved definition as the nodes are already registered. You'd have to delete all the node defs and abms and then register the plant anew. You may be able The fertility and description is already available through the nodes.

tl;dr: some benefits added with barely any code to maintain

:+1:

**Comment by sofar** [comment]
:+1: from me, but only if the array is documented in `game_api.txt`. This should be trivial: a casual mention of the array should suffice. @bas080 can you add this? Something like: `Registered plants are added to the .... array` around line 179.

**Comment by paramat** [comment]
@bas080 we'll merge this if you add docs as requested above.

**Comment by paramat** [comment]
I think we will have to add the docs ourselves on merge.

**Comment by paramat** [comment]
Applying: Store registered plant on namespace
error: patch failed: mods/farming/api.lua:1
error: mods/farming/api.lua: patch does not apply
Patch failed at 0001 Store registered plant on namespace

**Comment by rubenwardy** [comment]
78311a2
